invokeai/app/invocations/prompt.py

```diff
@@ -2,56 +2,52 @@
 from typing import Literal, Optional
 
 import numpy as np
-from pydantic import Field, validator
-
-from .baseinvocation import BaseInvocation, BaseInvocationOutput, InvocationConfig, InvocationContext
+from pydantic import validator
+
+from .baseinvocation import (
+    BaseInvocation,
+    BaseInvocationOutput,
+    InputField,
+    InvocationContext,
+    OutputField,
+    UIComponent,
+    UITypeHint,
+    title,
+    tags,
+)
 from dynamicprompts.generators import RandomPromptGenerator, CombinatorialPromptGenerator
 
 
 class PromptOutput(BaseInvocationOutput):
     """Base class for invocations that output a prompt"""
 
-    # fmt: off
     type: Literal["prompt"] = "prompt"
 
-    prompt: str = Field(default=None, description="The output prompt")
-    # fmt: on
-
-    class Config:
-        schema_extra = {
-            "required": [
-                "type",
-                "prompt",
-            ]
-        }
+    prompt: str = OutputField(description="The output prompt")
 
 
 class PromptCollectionOutput(BaseInvocationOutput):
     """Base class for invocations that output a collection of prompts"""
 
-    # fmt: off
     type: Literal["prompt_collection_output"] = "prompt_collection_output"
 
-    prompt_collection: list[str] = Field(description="The output prompt collection")
-    count: int = Field(description="The size of the prompt collection")
-    # fmt: on
-
-    class Config:
-        schema_extra = {"required": ["type", "prompt_collection", "count"]}
+    prompt_collection: list[str] = OutputField(
```

Comment:
I think we are a bit misguided in even having classes for `prompt` as something different from `string`. There's no functional difference, because what we pass to the prompt parser is a plain string. What do you think?

Reply:
Yeah, I agree with that. The compel prompts make sense since there's a lot of parsing in there

Reply:
I'll refactor all the prompt stuff into generic string stuff before putting this up for proper review, thanks for your feedback